src/coreclr/jit/target.h

```diff
@@ -208,8 +208,156 @@ enum _regMask_enum : unsigned
+
+#if defined(TARGET_AMD64) || defined(TARGET_ARM) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
 typedef unsigned __int64 regMaskTP;
+#elif defined(TARGET_ARM64)
+typedef unsigned __int64 regMaskSmall;
+struct regMaskTP
+{
+    unsigned __int64 low;
+
+    regMaskTP(unsigned __int64 regMask)
+        : low(regMask)
+    {
+    }
+
+    regMaskTP()
+    {
+    }
+
+    FORCEINLINE explicit operator bool() const
+    {
+        return low != 0;
+    }
+
+    FORCEINLINE explicit operator regMaskSmall() const
+    {
+        return (regMaskSmall)low;
+    }
+
+    FORCEINLINE explicit operator unsigned int() const
+    {
+        return (unsigned int)low;
+    }
+
+    FORCEINLINE unsigned __int64 getLow() const
+    {
+        return low;
+    }
+};
+
+static uint32_t PopCount(regMaskTP value)
+{
+    return BitOperations::PopCount(value.getLow());
+}
+
+static uint32_t BitScanForward(regMaskTP mask)
+{
+    return BitOperations::BitScanForward(mask.low);
+}
+
+static regMaskTP operator^(regMaskTP first, regMaskTP second)
+{
+    regMaskTP result(first.low ^ second.getLow());
+    return result;
+}
+
+static regMaskTP operator&(regMaskTP first, regMaskTP second)
+{
+    regMaskTP result(first.low & second.getLow());
+    return result;
+}
+
+static regMaskTP operator|(regMaskTP first, regMaskTP second)
+{
+    regMaskTP result(first.low | second.getLow());
+    return result;
+}
+
+static regMaskTP operator<<(regMaskTP first, const int b)
+{
+    regMaskTP result(first.low << b);
+    return result;
+}
+
+static regMaskTP operator>>(regMaskTP first, const int b)
+{
+    regMaskTP result(first.low >> b);
+    return result;
+}
+
+static regMaskTP& operator>>=(regMaskTP& first, const int b)
+{
+    first = first >> b;
+    return first;
+}
+
+static regMaskTP& operator|=(regMaskTP& first, regMaskTP second)
+{
+    first.low |= second.low;
+    return first;
+}
+
+static regMaskTP& operator^=(regMaskTP& first, regMaskTP second)
+{
+    first.low ^= second.low;
+    return first;
+}
+
+static regMaskSmall operator^=(regMaskSmall& first, regMaskTP second)
+{
+    first ^= second.low;
+    return first;
+}
+
+static regMaskSmall operator&=(regMaskSmall& first, regMaskTP second)
+{
+    first &= second.low;
+    return first;
+}
+
+static regMaskSmall operator|=(regMaskSmall& first, regMaskTP second)
+{
+    first |= second.low;
+    return first;
+}
+
+static regMaskTP& operator<<=(regMaskTP& first, regMaskTP second)
+{
+    first.low <<= second.low;
+    return first;
+}
+
+static regMaskTP& operator&=(regMaskTP& first, regMaskTP second)
+{
+    first.low &= second.low;
+    return first;
+}
+
+static bool operator==(regMaskTP first, regMaskTP second)
+{
+    return (first.low == second.low);
+}
+
+static bool operator!=(regMaskTP first, regMaskTP second)
+{
+    return (first.low != second.low);
+}
+
+static bool operator>(regMaskTP first, regMaskTP second)
+{
+    return (first.low > second.low);
+}
```

Comment:
I think we should get rid of this operator as well. There should be very few uses.